diff --git a/src/core/lv_obj_pos.c b/src/core/lv_obj_pos.c
--- a/src/core/lv_obj_pos.c
+++ b/src/core/lv_obj_pos.c
@@ -830,5 +830,12 @@ void lv_obj_get_transformed_area(const lv_obj_t * obj, lv_area_t * area, bool recursive,
     area->x2 = LV_MAX4(p[0].x, p[1].x, p[2].x, p[3].x);
     area->y1 = LV_MIN4(p[0].y, p[1].y, p[2].y, p[3].y);
     area->y2 = LV_MAX4(p[0].y, p[1].y, p[2].y, p[3].y);
-    lv_area_increase(area, 5, 5);
+    const lv_obj_t * current = obj;
+    while(current != NULL) {
+        if(_lv_obj_get_layer_type(current) == LV_LAYER_TYPE_TRANSFORM) {
+            lv_area_increase(area, 5, 5);
+            break;
+        }
+        current = recursive ? lv_obj_get_parent(current) : NULL;
+    }
 }
